### ISSUE TITLE
Add link label ?

### ISSUE THREAD

**Issue by lorico** [emiliorizzo/vue-d3-network#8]
Cool stuff Emilio !
Would it be possible to add a link "label" property & draw it ? That would be awesome.
Thanks!

**Comment by emiliorizzo** [comment]
Hi @lorico , thanks for your comments.
Yes is possible, maybe in next version.
In current version you can hook the link-click event to show link labels.

**Comment by lorico** [comment]
Thanks a lot for your answer !
I realized that there would be another potential enhancement that would make this lib amazing:
- ability to add "direction" to the edges (ie: add an arrow at the end of the tid) in SVG

Do you think that is possible ? 
Thanks again

**Comment by emiliorizzo** [comment]
@lorico 
Good idea.!
I'm working on it but I did not have time to finish.
Maybe next week.
Thanks for your contributions!

**Comment by lorico** [comment]
Amazing! You rock!

**Comment by emiliorizzo** [comment]
Hoi @lorico , see this example: https://codepen.io/emilio/pen/QOaQjP

**Comment by lorico** [comment]
Thanks a lot Emilio, that made the trick ! :)


Simon LEVASSEUR.

**Comment by emiliorizzo** [comment]
@lorico 
I think it's the simplest implementation, using the svg's features.
And you can draw different marks for each link.

**Comment by emiliorizzo** [comment]
@lorico , link labels: https://codepen.io/emilio/pen/ZampZb

**Comment by lorico** [comment]
Excellent !!!! Thanks a lot, I will implement that one on my side next week ! You're amazing !

**Comment by lorico** [comment]
Hi Emilio,
So it took me a bit longer than a week :)
I tested your example, but it doesn't work on my side:
- as soon as I draw the graph, I get eternal look of errors such as below 👍 
Any idea what is going on ? Looks like on some transitions, the attribute at least for circle & text doesn't get captured ?

NOTE: I was trying to combine your 2 helps of : 
- having a direction on the links (this one works alone)
- having on top the label on the links

Thanks !

**Error: <circle> attribute r: Expected length, "NaN".**
setAttr @ vue.js:4814
updateAttrs @ vue.js:4774
invokeCreateHooks @ vue.js:4256
createElm @ vue.js:4150
createChildren @ vue.js:4240
createElm @ vue.js:4148
createChildren @ vue.js:4240
createElm @ vue.js:4148
patch @ vue.js:4567
Vue._update @ vue.js:2639
updateComponent @ vue.js:2613
get @ vue.js:2936
Watcher @ vue.js:2928
Vue._mount @ vue.js:2612
Vue$3.$mount @ vue.js:6176
Vue$3.$mount @ vue.js:8547
init @ vue.js:1744
createComponent @ vue.js:4172
createElm @ vue.js:4115
createChildren @ vue.js:4240
createElm @ vue.js:4148
patch @ vue.js:4567
Vue._update @ vue.js:2639
updateComponent @ vue.js:2613
get @ vue.js:2936
Watcher @ vue.js:2928
Vue._mount @ vue.js:2612
Vue$3.$mount @ vue.js:6176
Vue$3.$mount @ vue.js:8547
init @ vue.js:1744
createComponent @ vue.js:4172
createElm @ vue.js:4115
createChildren @ vue.js:4240
createElm @ vue.js:4148
patch @ vue.js:4567
Vue._update @ vue.js:2639
updateComponent @ vue.js:2613
get @ vue.js:2936
Watcher @ vue.js:2928
Vue._mount @ vue.js:2612
Vue$3.$mount @ vue.js:6176
Vue$3.$mount @ vue.js:8547
init @ vue.js:1744
createComponent @ vue.js:4172
createElm @ vue.js:4115
patch @ vue.js:4567
Vue._update @ vue.js:2639
updateComponent @ vue.js:2613
get @ vue.js:2936
Watcher @ vue.js:2928
Vue._mount @ vue.js:2612
Vue$3.$mount @ vue.js:6176
Vue$3.$mount @ vue.js:8547
init @ vue.js:1744
createComponent @ vue.js:4172
createElm @ vue.js:4115
createChildren @ vue.js:4240
createElm @ vue.js:4148
addVnodes @ vue.js:4282
updateChildren @ vue.js:4411
patchVnode @ vue.js:4448
updateChildren @ vue.js:4364
patchVnode @ vue.js:4448
patch @ vue.js:4572
Vue._update @ vue.js:2646
updateComponent @ vue.js:2613
get @ vue.js:2936
run @ vue.js:3005
flushSchedulerQueue @ vue.js:2811
(anonymous) @ vue.js:477
nextTickHandler @ vue.js:426
Promise resolved (async)
timerFunc @ vue.js:441
queueNextTick @ vue.js:482
queueWatcher @ vue.js:2871
update @ vue.js:2995
notify @ vue.js:584
reactiveSetter @ vue.js:814
proxySetter @ vue.js:3336
(anonymous) @ app:265
j @ jquery-2.1.1.min.js:2
fireWith @ jquery-2.1.1.min.js:2
x @ jquery-2.1.1.min.js:4
(anonymous) @ jquery-2.1.1.min.js:4
XMLHttpRequest.send (async)
send @ jquery-2.1.1.min.js:4
ajax @ jquery-2.1.1.min.js:4
mounted @ app:255
callHook @ vue.js:2758
insert @ vue.js:1769
invokeInsertHook @ vue.js:4472
patch @ vue.js:4636
Vue._update @ vue.js:2639
updateComponent @ vue.js:2613
get @ vue.js:2936
Watcher @ vue.js:2928
Vue._mount @ vue.js:2612
Vue$3.$mount @ vue.js:6176
Vue$3.$mount @ vue.js:8547
Vue._init @ vue.js:3379
Vue$3 @ vue.js:3427
startApp @ app:347
stayHere @ login.js:65
AuthSuccess @ login.js:77
ajax.onreadystatechange @ login.js:142
XMLHttpRequest.send (async)
loginCheck @ login.js:113
setTimeout (async)
(anonymous) @ app:69

**14vue.js:4814 Error: <text> attribute x: Expected length, "NaN".**
setAttr @ vue.js:4814
updateAttrs @ vue.js:4774
invokeCreateHooks @ vue.js:4256
createElm @ vue.js:4150
createChildren @ vue.js:4240
createElm @ vue.js:4148
createChildren @ vue.js:4240
createElm @ vue.js:4148
patch @ vue.js:4567
Vue._update @ vue.js:2639
updateComponent @ vue.js:2613
get @ vue.js:2936
Watcher @ vue.js:2928
Vue._mount @ vue.js:2612
Vue$3.$mount @ vue.js:6176
Vue$3.$mount @ vue.js:8547
init @ vue.js:1744
createComponent @ vue.js:4172
createElm @ vue.js:4115
createChildren @ vue.js:4240
createElm @ vue.js:4148
patch @ vue.js:4567
Vue._update @ vue.js:2639
updateComponent @ vue.js:2613
get @ vue.js:2936
Watcher @ vue.js:2928
Vue._mount @ vue.js:2612
Vue$3.$mount @ vue.js:6176
Vue$3.$mount @ vue.js:8547
init @ vue.js:1744
createComponent @ vue.js:4172
createElm @ vue.js:4115
createChildren @ vue.js:4240
createElm @ vue.js:4148
patch @ vue.js:4567
Vue._update @ vue.js:2639
updateComponent @ vue.js:2613
get @ vue.js:2936
Watcher @ vue.js:2928
Vue._mount @ vue.js:2612
Vue$3.$mount @ vue.js:6176
Vue$3.$mount @ vue.js:8547
init @ vue.js:1744
createComponent @ vue.js:4172
createElm @ vue.js:4115
patch @ vue.js:4567
Vue._update @ vue.js:2639
updateComponent @ vue.js:2613
get @ vue.js:2936
Watcher @ vue.js:2928
Vue._mount @ vue.js:2612
Vue$3.$mount @ vue.js:6176
Vue$3.$mount @ vue.js:8547
init @ vue.js:1744
createComponent @ vue.js:4172
createElm @ vue.js:4115
createChildren @ vue.js:4240
createElm @ vue.js:4148
addVnodes @ vue.js:4282
updateChildren @ vue.js:4411
patchVnode @ vue.js:4448
updateChildren @ vue.js:4364
patchVnode @ vue.js:4448
patch @ vue.js:4572
Vue._update @ vue.js:2646
updateComponent @ vue.js:2613
get @ vue.js:2936
run @ vue.js:3005
flushSchedulerQueue @ vue.js:2811
(anonymous) @ vue.js:477
nextTickHandler @ vue.js:426
Promise resolved (async)
timerFunc @ vue.js:441
queueNextTick @ vue.js:482
queueWatcher @ vue.js:2871
update @ vue.js:2995
notify @ vue.js:584
reactiveSetter @ vue.js:814
proxySetter @ vue.js:3336
(anonymous) @ app:265
j @ jquery-2.1.1.min.js:2
fireWith @ jquery-2.1.1.min.js:2
x @ jquery-2.1.1.min.js:4
(anonymous) @ jquery-2.1.1.min.js:4
XMLHttpRequest.send (async)
send @ jquery-2.1.1.min.js:4
ajax @ jquery-2.1.1.min.js:4
mounted @ app:255
callHook @ vue.js:2758
insert @ vue.js:1769
invokeInsertHook @ vue.js:4472
patch @ vue.js:4636
Vue._update @ vue.js:2639
updateComponent @ vue.js:2613
get @ vue.js:2936
Watcher @ vue.js:2928
Vue._mount @ vue.js:2612
Vue$3.$mount @ vue.js:6176
Vue$3.$mount @ vue.js:8547
Vue._init @ vue.js:3379
Vue$3 @ vue.js:3427
startApp @ app:347
stayHere @ login.js:65
AuthSuccess @ login.js:77
ajax.onreadystatechange @ login.js:142
XMLHttpRequest.send (async)
loginCheck @ login.js:113
setTimeout (async)
(anonymous) @ app:69
14vue.js:4814

**Comment by emiliorizzo** [comment]
@lorico ?????  :open_mouth: :open_mouth: :open_mouth: :crying_cat_face:
If you want, provide a link to test.
Anyway the component is working... I don't understand your log error.

**Comment by lorico** [comment]
Ok I will give another try with a simple version to see what's wrong in
mine.
I'll keep you updated if I still have issue. Thanks a lot again !


Simon LEVASSEUR.

**Comment by emiliorizzo** [comment]
:+1:

**Comment by samul-1** [comment]
Hi, is there any way to have link labels behave like node labels?
For example, sometimes link labels will be turned upside down, or will rotate along with the link if a node is dragged around. Whereas node labels don't rotate around that way. Is there any way to "fix" link lables like that?

Also, what if I wanted to have the "name" attribute for the nodes and also add a second "label" attribute? So maybe I could have the "name" inside the node and the secondary label floating next to it much like the current "name" attribute behaves. That'd be great.
